src/clims/api/endpoints/substance_file_details.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import
+
+import posixpath
+
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from clims.services import NotFound
+from clims.api.serializers.models.organization_file import OrganizationFileSerializer
+try:
+    from django.http import (CompatibleStreamingHttpResponse as StreamingHttpResponse)
+except ImportError:
+    from django.http import StreamingHttpResponse
+
+
+class SubstanceFileDetailsEndpoint(OrganizationEndpoint):
+    permission_classes = (IsAuthenticated, )
+
+    def download(self, fp, name, size, content_type='application/octet-stream'):
+        response = StreamingHttpResponse(
+            iter(lambda: fp.read(4096), b''),
+            content_type=content_type
+        )
+        response['Content-Length'] = size
+        response['Content-Disposition'] = 'attachment; filename="%s"' % posixpath.basename(
+            " ".join(name.split())
+        )
+        return response
+
+    def download_from_model(self, model):
+        file = model.file
+        fp = file.getfile()
+        size = file.size
+        name = model.name
+        content_type = 'application/octet-stream'
+        # TODO: file.headers is an empty list now, not a dictionary(?) We should use the header
```

Comment:
I don't really understand this comment. IT looks like you are using and `octet-stream` in the response? No?

Reply:
What I meant there is that when a file is uploaded, one can set the content type of it. Then when downloading it (like here), we should use the content-type that was used when it was uploaded.

Here however, for some reason the content-type of the file is not set (headers was also unexpectedly a list). I assume it's probably my fault after some refactoring. This is not a huge issue at the moment, since we can just download these files as `application/octet-stream`, but I assume there may be cases where we wouldn't want that (although I don't see it at the moment).

Reply:
:+1: